scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/ParquetBucketMetadata.java

```diff
@@ -189,27 +192,18 @@ public void populateDisplayData(DisplayData.Builder builder) {
   }
 
   @Override
-  public boolean isPartitionCompatibleForPrimaryKey(BucketMetadata o) {
-    if (o == null || getClass() != o.getClass()) return false;
-    ParquetBucketMetadata<?, ?, ?> that = (ParquetBucketMetadata<?, ?, ?>) o;
-    return getKeyClass() == that.getKeyClass() && keyField.equals(that.keyField);
+  int hashPrimaryKeyMetadata() {
```

Comment:
those `hash` functions seem to be the same in all metadata implementations, aren't they? maybe push them down to the base class?

Reply:
They are the same in practice, but in theory, a `BucketMetadata` doesn't _have_ to have a `keyField`/`keyPath` and those aren't part of the base class, so we can't generalize it unfortunately :(

Reply:
Oh Ok!